sbtc/src/rpc.rs

```diff
@@ -175,3 +367,207 @@ impl BitcoinClient for BitcoinCoreClient {
+            witness: bitcoin::Witness::new(),
+        };
+        tx.input.push(deposit1);
+        tx.input.push(deposit2);
+
+        let fee = Amount::from_sat(500_000);
+
+        let tx_info = BitcoinTxInfo::from_tx(tx, fee);
+        let assessed_fee1 = tx_info.assess_input_fee(deposit_outpoint1).unwrap();
+        assert_eq!(assessed_fee1, fee / 2);
+
+        let assessed_fee2 = tx_info.assess_input_fee(deposit_outpoint2).unwrap();
+        assert_eq!(assessed_fee2, fee / 2);
+    }
+
+    #[test]
+    fn two_withdrawals_same_weight_split_the_fee() {
+        let mut tx = base_signer_transaction();
+        let locking_script = ScriptBuf::new_op_return([0; 10]);
+        let withdrawal = bitcoin::TxOut {
+            value: Amount::from_sat(250_000),
+            script_pubkey: ScriptBuf::new_p2sh(&locking_script.script_hash()),
+        };
+        tx.output.push(withdrawal.clone());
+        tx.output.push(withdrawal);
+        let fee = Amount::from_sat(500_000);
+
+        let tx_info = BitcoinTxInfo::from_tx(tx, fee);
+        let assessed_fee1 = tx_info.assess_output_fee(2).unwrap();
+        assert_eq!(assessed_fee1, fee / 2);
+
+        let assessed_fee2 = tx_info.assess_output_fee(3).unwrap();
+        assert_eq!(assessed_fee2, fee / 2);
+    }
+
+    #[test]
+    fn one_deposit_two_withdrawals_fees_add() {
+        // We're just testing that a "regular" bitcoin transaction,
+        // servicing a deposit and two withdrawals, will assess the fees in
+        // a normal way. Here we test that the fee is
+        let deposit_outpoint = OutPoint::new(Txid::from_byte_array([1; 32]), 0);
+
+        let mut tx = base_signer_transaction();
+        let deposit = bitcoin::TxIn {
+            previous_output: deposit_outpoint,
+            script_sig: ScriptBuf::new(),
+            sequence: bitcoin::Sequence::ZERO,
+            witness: bitcoin::Witness::new(),
+        };
+        tx.input.push(deposit);
+
+        let locking_script = ScriptBuf::new_op_return([0; 10]);
+        let withdrawal = bitcoin::TxOut {
+            value: Amount::from_sat(250_000),
+            script_pubkey: ScriptBuf::new_p2sh(&locking_script.script_hash()),
+        };
+        tx.output.push(withdrawal.clone());
+        tx.output.push(withdrawal);
+
+        let fee = Amount::from_sat(500_000);
+
+        let tx_info = BitcoinTxInfo::from_tx(tx, fee);
+        let input_assessed_fee = tx_info.assess_input_fee(deposit_outpoint).unwrap();
+        let output1_assessed_fee = tx_info.assess_output_fee(2).unwrap();
+        let output2_assessed_fee = tx_info.assess_output_fee(3).unwrap();
+
+        assert!(input_assessed_fee > Amount::ZERO);
+        assert!(output1_assessed_fee > Amount::ZERO);
+        assert!(output2_assessed_fee > Amount::ZERO);
+
+        let combined_fee = input_assessed_fee + output1_assessed_fee + output2_assessed_fee;
+
+        assert!(combined_fee >= fee);
+        // Their fees, in sats, should not add up to more than `fee +
```

Comment:
Wait, I'm confused... shouldn't the assessed fees add up to `fee`?

Reply:
No, because we use `div_ceil` when doing division in the `request.weight * fee / total_weight` calculation.

Reply:
Ah right and this is `/3` so it's rounding up. Gotcha :+1: